libyara/modules/pe.c

```diff
@@ -1961,6 +1999,7 @@ begin_declarations;
   declare_function("imports", "s", "i", imports_dll);
   declare_function("locale", "i", "i", locale);
   declare_function("language", "i", "i", language);
+  declare_function("rich_version", "i", "i", rich_version);
```

Comment:
Alright, I will try to do it later this evening.
On Thu, 26 Nov 2015 at 19:43, Victor M. Alvarez <notifications@github.com>
wrote:

> In libyara/modules/pe.c
> <https://github.com/plusvic/yara/pull/374#discussion_r46001914>:
>
> > @@ -1961,6 +1999,7 @@ begin_declarations;
> >    declare_function("imports", "s", "i", imports_dll);
> >    declare_function("locale", "i", "i", locale);
> >    declare_function("language", "i", "i", language);
> > +  declare_function("rich_version", "i", "i", rich_version);
>
> Yes, a pair of functions like this:
>
> rich_signature.version(version, toolid [optional])
> rich_signature.toolid(toolid, version [optional])
>
> This way you can check for any combination of toolid/version.
>
> This would requiere actually four functions, two overloaded "version"
> functions and two overloaded "toolid" functions. I suggest writing a single
> internal function receiving both a toolid and a version and accepting a
> special value as a wildcard for either toolid or version. The other 4
> functions could invoke the internal function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/plusvic/yara/pull/374/files#r46001914>.
>